### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by blase147**
In this PR, I did the following:

Solved N+1 problem when fetching all posts and their comments for a user without using the bullet gem

Used Capybara to write integration tests for each view of the Blog App project according to the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md)

Implemented tests for user index page
Implemented tests for user show page
Implemented tests for post index page
Implemented tests for post show page

### CODE REVIEW

File: spec/integration/post_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    end
+
+    it 'displays the comment text' do
+      visit user_post_path(@user, @post.id)
+      expect(page).to have_content('I can\'t do it')
+      expect(page).to have_content('Nothing is impossible')
+      expect(page).to have_content('Thanks')
+    end
+  end
+end
```
Comment (rotimiazeez) [inline spec/integration/post_show_spec.rb]: - Please note that the post-show specs are currently failing. Kindly make sure all your tests are passing to pass this milestone 😉 
Screenshot from 2023-02-20 13-36-59

File: spec/integration/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      click_link(@post.text)
+      expect(page).to have_current_path(user_post_path(@user, @post))
+    end
+
+    it 'redirects to open all posts of a user' do
+      click_link('See all Posts')
+      expect(page).to have_current_path(user_posts_path(@user))
+    end
+  end
+end
```
Comment (rotimiazeez) [inline spec/integration/user_show_spec.rb]: - Please note that the user-show specs are currently failing. Kindly make sure all your tests are passing to pass this milestone 😉 
Screenshot from 2023-02-20 13-41-36

File: spec/integration/user_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        expect(page).to have_content("Number of posts: #{user.PostsCounter}")
+      end
+    end
+
+    it "is redirected to that user's show page" do
+      click_link(@user.name)
+      expect(page).to have_current_path(user_path(@user.id))
+    end
+  end
+end
```
Comment (rotimiazeez) [inline spec/integration/user_index_spec.rb]: - Please note that the user-index specs are currently failing. Kindly make sure all your tests are passing to pass this milestone 😉 
Screenshot from 2023-02-20 13-44-33

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @blase147 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

**Review by sumairq — APPROVED** [review]
### Hi @blase147 , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**